src/sentry/workflow_engine/migrations/0012_migrate_metric_alerts.py

```diff
@@ -0,0 +1,187 @@
+# Generated by Django 5.1.1 on 2024-11-08 23:16
+
+from enum import Enum, IntEnum
+
+from django.apps.registry import Apps
+from django.db import migrations, models
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBarApprox
+
+
+class PriorityLevel(IntEnum):
+    LOW = 25
+    MEDIUM = 50
+    HIGH = 75
+
+
+class DetectorPriorityLevel(IntEnum):
+    OK = 0
+    LOW = PriorityLevel.LOW
+    MEDIUM = PriorityLevel.MEDIUM
+    HIGH = PriorityLevel.HIGH
+
+
+class IncidentStatus(Enum):
+    OPEN = 1
+    CLOSED = 2
+    WARNING = 10
+    CRITICAL = 20
+
+
+class DataSourceType(models.IntegerChoices):
+    SNUBA_QUERY_SUBSCRIPTION = 1
+    SNUBA_QUERY = 2
+
+
+class DataConditionGroupType(models.TextChoices):
+    ANY = "any"
+    ALL = "all"
+    NONE = "none"
+
+
+def migrate_metric_alerts(apps: Apps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    AlertRule = apps.get_model("sentry", "AlertRule")
+    AlertRuleProjects = apps.get_model("sentry", "AlertRuleProjects")
+    AlertRuleTrigger = apps.get_model("sentry", "AlertRuleTrigger")
+    AlertRuleTriggerAction = apps.get_model("sentry", "AlertRuleTriggerAction")
+    # AlertRuleActivity = apps.get_model("sentry", "AlertRuleActivity")
+    QuerySubscription = apps.get_model("sentry", "QuerySubscription")
+    Incident = apps.get_model("sentry", "Incident")
+
+    DataSource = apps.get_model("sentry", "DataSource")
+    DataConditionGroup = apps.get_model("sentry", "DataConditionGroup")
+    Workflow = apps.get_model("sentry", "Workflow")
+    Detector = apps.get_model("sentry", "Detector")
+    DetectorState = apps.get_model("sentry", "DetectorState")
+    DataCondition = apps.get_model("sentry", "DataCondition")
+    Action = apps.get_model("sentry", "Action")
+    DataConditionGroupAction = apps.get_model("sentry", "DataConditionGroupAction")
+    DataSourceDetector = apps.get_model("sentry", "DataSourceDetector")
+    DetectorWorkflow = apps.get_model("sentry", "DetectorWorkflow")
+    WorkflowDataConditionGroup = apps.get_model("sentry", "WorkflowDataConditionGroup")
+
+    for rule in RangeQuerySetWrapperWithProgressBarApprox(AlertRule.objects.all()):
+        # should I skip migrating snapshotted and disabled rules?
+        # what about the date_added and date_updated fields? can I write to those to preserve history?
+        # description column will be added to the Detector model, but at the time of writing it isn't there yet. will need to update this
+        # same for threshold_period, will likely be added to the Detector model but isn't there yet
+        # AlertRuleActivity creation rows need to be backfilled into the Detector model, needs new created_by and created_at columns
```

Comment:
Not sure if we need this if we're not keeping history?

We probably do need to migrate the audit logs over though...

Reply:
If we want to maintain the data about which user created the rule and when we'll need to put this somewhere. The audit log table has a lot of this but since it was added more recently there's a ~2 year gap of rule data that's missing. I wouldn't be opposed to trying to backfill that table with this instead though.